src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -0,0 +1,318 @@
+using GraphQL.Subscription;
+using GraphQL.Types;
+using GraphQLParser.AST;
+
+namespace GraphQL.Execution
+{
+    /// <summary>
+    /// Executes a subscription.
+    /// </summary>
+    public class SubscriptionExecutionStrategy : ExecutionStrategy
+    {
+        private readonly IExecutionStrategy _baseExecutionStrategy;
+
+        /// <summary>
+        /// Initializes a new instance with a parallel execution strategy for child nodes.
+        /// </summary>
+        public SubscriptionExecutionStrategy()
+            : this(new ParallelExecutionStrategy()) // new instance of shared variables within ParallelExecutionStrategy; do not use ParallelExecutionStrategy.Instance
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance with the specified execution strategy for child nodes.
+        /// </summary>
+        public SubscriptionExecutionStrategy(IExecutionStrategy baseExecutionStrategy)
+        {
+            _baseExecutionStrategy = baseExecutionStrategy ?? throw new ArgumentNullException(nameof(baseExecutionStrategy));
+        }
+
+        /// <summary>
+        /// Gets a static instance of <see cref="SubscriptionExecutionStrategy"/>.
+        /// </summary>
+        public static SubscriptionExecutionStrategy Instance { get; } = new();
+
+        /// <summary>
+        /// Executes a GraphQL subscription request and returns the result. The result consists
+        /// of one or more streams of GraphQL responses, returned within <see cref="SubscriptionExecutionResult.Streams"/>.
+        /// No serializable <see cref="ExecutionResult"/> is directly returned.
```

Comment:
> No serializable <see cref="ExecutionResult"/> is directly returned.

What does it mean? Method does return ExecutionResult.

Reply:
This is an old comment and is partially correct and partially incorrect.  For a successful execution at this point, there is no serialized response given across the WebSocket interface.  Only upon processing and serialization of the first data packet does data start traveling to the client.

However, in the case of an error, the `ExecutionResult` does contain error information that would be provided to the client, along with a failure notice indicating the attempt to establish a subscription has failed.

With the older `graphql-ws` protocol, the execution result is serialized in the case of an error.

With the newer `graphql-transport-ws` protocol, only the errors (as an array) is serialized in the case of an error.

So this comment really relates more to the protocol in use (defined in the server project) than the response here.

Reply:
```suggestion
        /// No serializable <see cref="ExecutionResult"/> is directly returned unless an error has occurred.
```

Reply:
OK?